pisces/models.py

```diff
@@ -25,42 +27,123 @@
 from sklearn.model_selection import LeaveOneOut
 from sklearn.preprocessing import StandardScaler
 from concurrent.futures import ProcessPoolExecutor
+from sklearn.ensemble import RandomForestClassifier
 from .data_sets import DataSetObject, ModelInput1D, ModelInputSpectrogram, ModelOutputType, DataProcessor
 
 # %% ../nbs/02_models.ipynb 6
 class SleepWakeClassifier(abc.ABC):
-    """
+    """ Abstract class for sleep/wake classifiers. 
     """
     @abc.abstractmethod
-    def get_needed_X_y(self, data_set: DataSetObject, id: str) -> Tuple[np.ndarray, np.ndarray] | None:
-        pass
-    def train(self, examples_X: List[pl.DataFrame] = [], examples_y: List[pl.DataFrame] = [], 
-              pairs_Xy: List[Tuple[pl.DataFrame, pl.DataFrame]] = [], 
-              epochs: int = 10, batch_size: int = 32):
-        pass
+    def __init__(self, model=None, data_processor=None):
+        self.model = model
+        self.scaler = StandardScaler()
+        self.pipeline = Pipeline([('scaler', self.scaler), ('model', self.model)])
```

Comment:
I like how this reduces the code we need to repeat in subclasses, but it does make `self.model`, `self.pipeline` and `self.scaler` a part of this class. 

However, these are not present in the `MOResUNetPretrained` subclass; I don't think they even exist for that class. So, someone could write code using these, and not realize that code will crash when `MOResUNetPretrained` is used.

What I would recommend is:
1. changing `MOResUNetPretrained` so that `.tf_model` becomes `.model`
2. removing `self.scaler` entirely, and replace this line with `self.pipeline = Pipeline([('scaler', StandardScaler()), ('model', self.model)])`
3. Is there some way we can still use a `Pipeline` with TF models? They have a `.fit` method that AFAIK implements the SKLearn API. Or, come up with a custom value for `self.pipeline` in the subclass that is legit?

Doing these can also help us re-use code currently split into 2 that's actually just two approaches to the train=>evaluate problem.

Reply:
I've been thinking about how we can get both simple models and MO under a single train function but I think it will require a whole refactoring because the `train` for simple models and the one from MO are quite different in the inputs they require and also the output. Also, I'm don't think we want a scaler for MO since we take care of all the pre-processing and scaling in a separate step. Maybe we should move that refactoring to a new issue/PR?

I did on this PR though update MO's `init` and changed `tf_model` for `model`.